### DISCUSSION TITLE
When a user wants to delete a dataset , the confirmation message could be more shaped

### DISCUSSION THREAD

**Discussion by xavier-GitHub76** [apache/superset#29646] [Ideas]
Hello,

when a user wants to delete a dataset , the following confirmation message is displayed
We can see on the example that no element is linked to the dataset.
When elements are linked, a list could be displayed to identify elements
When elements are not linked, a simpler message could be displayed to inform user

## Screenshot
Before :
![image](https://github.com/user-attachments/assets/c46fa11f-c046-4afe-a65b-cbd6dace2104)

After : 
If no link : 
![image](https://github.com/user-attachments/assets/1de984c1-fc47-4b12-b93e-cb93ac2564bb)

If links : 
![image](https://github.com/user-attachments/assets/e3cee3ed-30db-451e-8681-389311c85a3c)

Best regards

**Comment by rusackas** [comment]
I love the idea! It's more enhancement than bug, so we might move it to a Github Ideas discussion. If you're willing to implement this, we'd be more than happy to review a PR. Otherwise, I'll take @kasiazjc here for design input and @yousoph to see if anyone wants to adopt the idea.

**Comment by xavier-GitHub76** [comment]
I don't see how convert the issue in discussion. 
Should I manually close the bug and create a discussion ?

**Comment by kasiazjc** [comment]
I think it's a great idea too! I can a case that needs to be covered - if dataset is linked to more than ~5 charts (which probably is a case most times) the modal would get super long and overwhelming. Maybe the solution would be to list let's say 5 most recent add something like "...and [x] others]". We could display the tooltip with the rest of the charts, or just leave as is with information that you can go to  charts crud view, filter via the dataset and asses the charts?
